packages/react-ds/src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useState, useEffect, useCallback } from 'react'
+
+import { IPagination } from '@venice/core/models'
+import styles from '@venice/styles/components/Pagination.module.scss'
+
+import {
+  getRangeIndexes,
+  hasNextRange,
+  hasPreviousRange,
+} from './helpers/PaginationHelper'
+// TODO: these icons should be replaced when there is an icon library
+import AngleLeft from './icons/AngleLeft'
+import AngleRight from './icons/AngleRight'
+import PaginationRange from './PaginationRange'
+
+const Pagination: React.FC<IPagination> = ({
+  count = 1,
+  page = 1,
+  isLoading = false,
+  onChange = () => null,
+}: IPagination) => {
+  const [isMobile, setIsMobile] = useState(false)
+
+  useEffect(() => {
+    let doit: NodeJS.Timeout
+    const timeOut = 200
+    const maxPhoneWidth = 420
+
+    const checkDevice = () => setIsMobile(window.innerWidth <= maxPhoneWidth)
+    checkDevice()
+
+    const onResize = () => {
+      clearTimeout(doit)
+      doit = setTimeout(() => checkDevice(), timeOut)
```

Comment:
should be a debounce function instead of timeout. Timeout in this case don't change behaviour, just delay it